rootfs/scheduler/__init__.py

```diff
@@ -639,29 +639,40 @@ def _set_image_secret(self, data, namespace, **kwargs):
         Take registry information and set as an imagePullSecret for an RC / Deployment
         http://kubernetes.io/docs/user-guide/images/#specifying-imagepullsecrets-on-a-pod
         """
+        secret_name = settings.REGISTRY_SECRET_PREFIX
         registry = kwargs.get('registry', {})
-        if not registry:
+        if not registry and settings.REGISTRY_LOCATION == 'on-cluster':
+            return
+        elif registry:
+            # try to get the hostname information
+            hostname = registry.get('hostname', None)
+            if not hostname:
+                hostname, _ = docker_auth.split_repo_name(kwargs.get('image'))
+            username = registry.get('username')
+            password = registry.get('password')
+        elif settings.REGISTRY_LOCATION == 'off-cluster':
+            secret = self.get_secret('deis', 'registry-secret').json()
+            username = secret['data']['username']
+            password = secret['data']['password']
+            hostname = secret['data']['server']
+            if hostname == '':
+                hostname = "https://index.docker.io/v1/"
```

Comment:
hostname is hardcoded only if user doesn't specify it where in we assume it to be dockerhub.